### ISSUE TITLE
Flutter analyze is not working

### ISSUE THREAD

**Issue by JCKodel** [flutter/flutter#63566]
## Steps to Reproduce

Flutter run reports errors that flutter analyze don't (so VSCode also reports 0 errors in code)

<details><summary>base_ads.dart</summary>

```dart
import 'dart:async';

import 'package:flutter/foundation.dart';

/* —————————————————————————————————————————————————————————————————————————— */
abstract class BaseAds {
  BaseAds(String adUnitId) : _adUnitId = adUnitId;

  final String _adUnitId;
  String get adUnitId => _adUnitId;

  bool _isLoaded = false;
  bool get isLoaded => _isLoaded;

  Completer<bool> _currentCompleter;

/* —————————————————————————————————————————————————————————————————————————— */
  Future<void> reload() async {
    _isLoaded = false;
    onLoad();
  }

/* —————————————————————————————————————————————————————————————————————————— */
  Future<bool> show() {
    _currentCompleter = Completer<bool>();

    Timer(
      const Duration(milliseconds: 500),
      () {
        if (_currentCompleter.isCompleted == false) {
          _currentCompleter.complete(false);
        }
      },
    );

    onShow();

    return _currentCompleter.future;
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @protected
  void onLoaded() {
    _isLoaded = true;
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @protected
  void onFailedToLoad() {
    _isLoaded = false;
    Timer(const Duration(seconds: 15), () => reload());

    if (_currentCompleter.isCompleted == false) {
      _currentCompleter.complete(false);
    }
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @protected
  void onClicked();

/* —————————————————————————————————————————————————————————————————————————— */
  @protected
  void onOpened();

/* —————————————————————————————————————————————————————————————————————————— */
  @protected
  @mustCallSuper
  void onClosed() {
    _isLoaded = false;
    _currentCompleter.complete(true);
  }

  void onShow();
  void onLoad();
}
```
</details>

<details><summary>base_rewarded_ads.dart</summary>

```dart
import 'package:flutter/foundation.dart';

import '../business_service.dart';
import '../remote_config_service.dart';
import 'base_ads.dart';

/* —————————————————————————————————————————————————————————————————————————— */
abstract class BaseRewardedAds extends BaseAds {
  BaseRewardedAds(String adUnitId) : super(adUnitId);

  bool _rewarded = false;
  bool _clicked = false;
  final _stopwatch = Stopwatch();

  @override
  @mustCallSuper
  void onLoad() {
    _rewarded = false;
    _clicked = false;
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @override
  void onOpened() {
    _rewarded = false;
    _clicked = false;
    _stopwatch.reset();
    _stopwatch.start();
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @protected
  void onRewarded() {
    _rewarded = true;
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @override
  @mustCallSuper
  void onClicked() {
    _clicked = true;
  }

/* —————————————————————————————————————————————————————————————————————————— */
  void onCompleted() {
    if (_rewarded == false) {
      return;
    }

    final diamonds = (_stopwatch.elapsed.inSeconds.clamp(1, 30) / RemoteConfigService.instance.diamondsPer30Seconds).clamp(1, 9999);

    BusinessService.instance.addDiamondBalance(_clicked ? diamonds * 2 : diamonds);
    _rewarded = false;
    _clicked = false;
  }
}
```
</details>

<details><summary>ad_mob_rewarded_ads.dart</summary>

```dart
import 'package:firebase_admob/firebase_admob.dart';

import '../../logger.dart';
import 'base_rewarded_ads.dart';

/* —————————————————————————————————————————————————————————————————————————— */
class AdMobRewardedAds extends BaseRewardedAds {
  AdMobRewardedAds._(String adUnitId) : super(adUnitId);

  static AdMobRewardedAds _instance;
  static AdMobRewardedAds get instance => _instance;

/* —————————————————————————————————————————————————————————————————————————— */
  static void initialize(String adUnitId) {
    _instance = AdMobRewardedAds._(adUnitId);

    RewardedVideoAd.instance.listener = (RewardedVideoAdEvent event, {int rewardAmount, String rewardType}) {
      Logger.info("AdMobRewarded ${event}");

      switch (event) {
        case RewardedVideoAdEvent.loaded:
          onLoaded();
          break;
        case RewardedVideoAdEvent.failedToLoad:
          onFailedToLoad();
          break;
        case RewardedVideoAdEvent.closed:
          onClosed();
          break;
        case RewardedVideoAdEvent.rewarded:
          onRewarded();
          break;
        case RewardedVideoAdEvent.started:
          onOpened();
          break;
        case RewardedVideoAdEvent.completed:
          onCompleted();
          break;
        case RewardedVideoAdEvent.leftApplication:
          onClicked();
          break;
        default:
      }
    };
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @override
  void onLoad() {
    super.onLoad();

    RewardedVideoAd.instance.load(adUnitId: adUnitId).then(
      (result) {
        if (result == false) {
          onFailedToLoad();
        }
      },
    );
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @override
  void onShow() {
    if (isLoaded) {
      RewardedVideoAd.instance.show().then(
        (result) {
          if (result == false) {
            onFailedToLoad();
          }
        },
      );
    }
  }

/* —————————————————————————————————————————————————————————————————————————— */
  @override
  void onClosed() {
    super.onClosed();

    RewardedVideoAd.instance.load(adUnitId: adUnitId).then(
      (result) {
        if (result == false) {
          onFailedToLoad();
        }
      },
    );
  }
}

```
</details>

## Logs

```
C:\Projects\MCC2020\Frontend>flutter analyze
Analyzing Frontend...
No issues found! (ran in 2.0s)

C:\Projects\MCC2020\Frontend>flutter build bundle
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:22:11: Error: Method not found: 'onLoaded'.
          onLoaded();
          ^^^^^^^^
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:25:11: Error: Method not found: 'onFailedToLoad'.
          onFailedToLoad();
          ^^^^^^^^^^^^^^
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:28:11: Error: Method not found: 'onClosed'.
          onClosed();
          ^^^^^^^^
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:31:11: Error: Method not found: 'onRewarded'.
          onRewarded();
          ^^^^^^^^^^
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:34:11: Error: Method not found: 'onOpened'.
          onOpened();
          ^^^^^^^^
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:37:11: Error: Method not found: 'onCompleted'.
          onCompleted();
          ^^^^^^^^^^^
lib/kfx/services/ads/ad_mob_rewarded_ads.dart:40:11: Error: Method not found: 'onClicked'.
          onClicked();
          ^^^^^^^^^
Target kernel_snapshot failed: Exception
Failed to build bundle.

C:\Projects\MCC2020\Frontend>
```

```
[✓] Flutter (Channel stable, 1.20.0, on Microsoft Windows [Version 10.0.19042.423], locale pt-BR)
    • Flutter version 1.20.0 at C:\flutter
    • Framework revision 840c9205b3 (7 days ago), 2020-08-04 20:55:12 -0700
    • Engine revision c8e3b94853
    • Dart version 2.9.0


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\jckod\AppData\Local\Android\Sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = C:\Users\jckod\AppData\Local\Android\Sdk
    • ANDROID_SDK_ROOT = C:\Users\jckod\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.47.3)
    • VS Code at C:\Users\jckod\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • XT1033 (mobile)                    • 0423490246      • android-arm   • Android 5.0.2 (API 21)
    • SM J415G (mobile)                  • 785dd05b        • android-arm   • Android 9 (API 28)
    • ASUS Z00UD (mobile)                • G7AZB6026545GU7 • android-arm64 • Android 6.0.1 (API 23)
    • Android SDK built for x86 (mobile) • emulator-5554   • android-x86   • Android 10 (API 29) (emulator)

• No issues found!
```

VSCode reports 0 issues with the code (no errors, nothing on Problems tab). Errors just appear when hitting F5.

**Comment by JCKodel** [comment]
The error is trying to call a member of the super class inside a static method. 
The bug here is that the error is not reported at all.

Just for comparison, in C#, that would throw "An object reference is required for the non-static field, method, or property 'Program.foo()'". *THIS* is a worthy compiler error =)

**Comment by darshankawar** [comment]
Hi @JCKodel,
I tried to replicate this on latest Stable (1.20.1) with your code snippet, but seeing `flutter analyze` indicating errors:

```
  error • Local variable 'onShow' can't be referenced before it is declared • lib/main.dart:6589:3 • referenced_before_declaration
  error • This expression has a type of 'void' so its value can't be used • lib/main.dart:6589:3 • use_of_void_result
   info • The declaration 'onLoaded' isn't referenced • lib/main.dart:6596:8 • unused_element
```
Is there any folder structure to follow for the files you shared in order to replicate the issue ?
Thanks.

**Comment by JCKodel** [comment]
> Is there any folder structure to follow for the files you shared in order to replicate the issue ?

I tried to replicate it, but I couldn't (Flutter 1.20.1). Either was a problem fixed after machine shutdown or by Flutter 1.20.1.
